### PR TITLE
addred link to  Go to direct deposit to manage your Montgomery Gl Bil…

### PR DESCRIPTION
**Pull request by wafimohamed**
…l benefit

            payment address

**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary
As: 

VYE Dev

I need: 

Add feature toggles to the React widget and the Profile address and direct deposit forms

So that:  

Once VYE is in Prod, we can turn off the forms in VYE and turn on the forms in the Profile.

Considerations: 

NA

Acceptance Criteria:  

Enter the acceptance criteria that will be used to test the user story.  

Feature toggle vyeLoginWidget is in vets-website
Feature toggle toggleVyeAddressDirectDepositFormsInProfile is in vets-website
Testing Considerations: 

NA, this is adding feature toggles to the FeaturFlagNames.json file.

Other Considerations:  

NA

Enter any other items that should be considered, such as whether as RFAC is needed, technical items, etc. 
- _(Summarize the changes that have been made to the platform)_
- _(If bug, how to reproduce)_
- _(What is the solution, why is this the solution)_
- _(Which team do you work for, does your team own the maintenance of this component?)_
- _(If using a flipper, what is the end date of the flipper being required/success criteria being targeted)_

## Related issue(s)

- _Link to ticket created in va.gov-team repo_ https://jira.devops.va.gov/browse/VYE-420
department-of-veterans-affairs/va.gov-team#0000
- _Link to previous change of the code/bug (if applicable)_
department-of-veterans-affairs/vets-website#0000
- _Link to epic if not included in ticket_
department-of-veterans-affairs/va.gov-team#0000

## Testing done

- _Describe what the old behavior was prior to the change_
 added feature toggle for mailing address and direct Deposit in the profile page. 
- _Describe the steps required to verify your changes are working as expected_
- _Describe the tests completed and the results_
- _Exclusively stating 'Specs and automated tests passing' is NOT acceptable as appropriate testing
- _Optionally, provide a link to your [test plan](https://depo-platform-documentation.scrollhelp.site/developer-docs/create-a-test-plan-in-testrail) and [test execution records](https://depo-platform-documentation.scrollhelp.site/developer-docs/execute-tests-in-testrail)_

## Screenshots
image
image
image
A We can't confirm the address you

image

image
image
image
image

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |        |       |
| Desktop |        |       |

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by jsimonVA — DISMISSED** [review]

**Review by jsimonVA — DISMISSED** [review]

File: src/platform/utilities/feature-toggles/featureFlagNames.json [hunk 1]
```diff
@@ -256,5 +256,7 @@
   "virtualAgentEnableMsftPvaTesting": "virtual_agent_enable_msft_pva_testing",
   "virtualAgentEnableNluPvaTesting": "virtual_agent_enable_nlu_pva_testing",
   "yellowRibbonAutomatedDateOnSchoolSearch": "yellow_ribbon_automated_date_on_school_search",
-  "toggleVyeAdressDirectDepositForms": "toggle_vye_adress_direct_deposit_forms"
+  "toggleVyeAdressDirectDepositForms": "toggle_vye_adress_direct_deposit_forms",
+  "toggleVyeAdressDirectDepositFormsInProfile": "toggle_vye_adress_direct_deposit_forms_in_profile",
```
Comment (adamwhitlock1) [inline src/platform/utilities/feature-toggles/featureFlagNames.json:260]: There is a typo in the toggle names here: adress should be address

File: src/applications/personalization/profile/components/contact-information/addresses/AddressesTable.jsx [hunk 2]
```diff
@@ -8,24 +8,36 @@ import {
...
+        {toggleValue ? (
+          <NavLink
+            activeClassName="is-active"
+            exact
+            to="/profile/direct-deposit"
+          >
+            Go to direct deposit to manage your Montgomery Gl Bill benefit
+            payment address.
+          </NavLink>
+        ) : null}
```
Comment (adamwhitlock1) [inline src/applications/personalization/profile/components/contact-information/addresses/AddressesTable.jsx:39]: You don't need to have a ternary here to do conditional rendering.

```
        {toggleValue && (
          <NavLink
            activeClassName="is-active"
            exact
            to="/profile/direct-deposit"
          >
            Go to direct deposit to manage your Montgomery Gl Bill benefit
            payment address.
          </NavLink>
        ) }
        ```

**Review by jsimonVA — DISMISSED** [review]

File: src/platform/utilities/feature-toggles/featureFlagNames.json [hunk 3]
```diff
@@ -256,5 +256,7 @@
   "virtualAgentEnableMsftPvaTesting": "virtual_agent_enable_msft_pva_testing",
   "virtualAgentEnableNluPvaTesting": "virtual_agent_enable_nlu_pva_testing",
   "yellowRibbonAutomatedDateOnSchoolSearch": "yellow_ribbon_automated_date_on_school_search",
-  "toggleVyeAdressDirectDepositForms": "toggle_vye_adress_direct_deposit_forms"
+  "toggleVyeAdressDirectDepositForms": "toggle_vye_adress_direct_deposit_forms",
```
Comment (adamwhitlock1) [inline src/platform/utilities/feature-toggles/featureFlagNames.json:259]: 'toggle_vye_adress_direct_deposit_forms' also has the typo for `address`. Is this in use already by your team?

Comment (wafimohamed) [reply src/platform/utilities/feature-toggles/featureFlagNames.json:259]: yeah I can change that in other PR for my team. I have changed the one which is used for the profile

Comment (wafimohamed) [reply src/platform/utilities/feature-toggles/featureFlagNames.json:259]: I have updated in this pr https://github.com/department-of-veterans-affairs/vets-website/pull/30218

File: src/platform/utilities/feature-toggles/featureFlagNames.json [hunk 4]
```diff
@@ -256,5 +256,7 @@
...
+  "toggleVyeAddressDirectDepositFormsInProfile": "toggle_vye_adress_direct_deposit_forms_in_profile",
```
Comment (adamwhitlock1) [inline src/platform/utilities/feature-toggles/featureFlagNames.json:260]: Looks like you fixed the left side key spelling, but the value string still has the typo.

```suggestion
  "toggleVyeAddressDirectDepositFormsInProfile": "toggle_vye_address_direct_deposit_forms_in_profile",
```

**Review by adamwhitlock1 — CHANGES_REQUESTED** [review]

File: src/platform/utilities/feature-toggles/featureFlagNames.json (hunk 4 above)
Comment (wafimohamed) [reply src/platform/utilities/feature-toggles/featureFlagNames.json:260]: Oops need to fix that in the vets-api repo first 😖

**Review by jsimonVA — DISMISSED** [review]

File: src/applications/personalization/profile/components/direct-deposit/vye/components/addressSchema.js [hunk 5]
```diff
@@ -30,13 +30,7 @@ export const getFormSchema = (defaultVeteranName, formData = {}) => {
...
         type: 'object',
         properties: {},
-        default: formData['view:livesOnMilitaryBas'],
-      },
-      fullName: {
-        type: 'string',
-        minLength: 1,
-        maxLength: 100,
-        default: defaultVeteranName,
+        default: !formData['view:livesOnMilitaryBas'],
```
Comment (adamwhitlock1) [inline src/applications/personalization/profile/components/direct-deposit/vye/components/addressSchema.js:33]: missing an 'e' at the end of this. Should be `view:livesOnMilitaryBase`

File: src/applications/personalization/profile/components/direct-deposit/vye/components/addressSchema.js [hunk 6]
```diff
@@ -106,19 +100,20 @@ export const getFormSchema = (defaultVeteranName, formData = {}) => {
...
     'view:livesOnMilitaryBaseInfo': {
       'ui:description': () => (
-        <p className="profile-military-domestic">
-          U.S. military bases are considered a domestic address and a part of
-          the United States.
-        </p>
+        <va-additional-info trigger="Learn more about military base addresses">
+          <p>
+            We automatically enter the United States as your country if you live
+            on a military case outside of the country.
```
Comment (adamwhitlock1) [inline src/applications/personalization/profile/components/direct-deposit/vye/components/addressSchema.js:110]: it should be `military base` and not `military case`

**Comment by wafimohamed** [comment]
This PR has a lot of failing tests

**Pull request closed without merging**